File: Library.Core.Web/src/app/view-models/publication-house/getPublication-houseView.ts
export class GetPublicationHouseView {
  constructor(
    public publicationHouses: PublicationHouseGetPublicationHouseViewItem[],
    public allBooks: BookGetPublicationHouseViewItem[]
  ) { }
}

export class PublicationHouseGetPublicationHouseViewItem {
  constructor(
    public id: number = 0,
    public name?: string,
    public adress?: string,
    public books?: BookGetPublicationHouseViewItem[]
  ) { }
}

export class BookGetPublicationHouseViewItem {
  constructor(
    public id: number = 0,
    public name?: string,
    public yearOfPublishing: number = 0
  ) { }
}

File: Library.Core.ViewModelLayer/ViewModels/PublicationHouseViewModels/GetPublicationHouseView.cs
using System.Collections.Generic;

namespace Library.Core.ViewModelLayer.ViewModels.PublicationHouse
{
    public class GetPublicationHouseView
    {
        public ICollection<PublicationHouseGetPublicationHouseViewItem> PublicationHouses { get; set; }
        public ICollection<BookGetPublicationHouseViewItem> AllBooks { get; set; }

        public GetPublicationHouseView()
        {
            PublicationHouses = new List<PublicationHouseGetPublicationHouseViewItem>();
            AllBooks = new List<BookGetPublicationHouseViewItem>();
        }
    }

    public class PublicationHouseGetPublicationHouseViewItem
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string Adress { get; set; }

        public virtual ICollection<BookGetPublicationHouseViewItem> Books { get; set; }

        public PublicationHouseGetPublicationHouseViewItem()
        {
            Books = new List<BookGetPublicationHouseViewItem>();
        }
    }

    public class BookGetPublicationHouseViewItem
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public int YearOfPublishing { get; set; }

        public BookGetPublicationHouseViewItem()
        {

        }
    }
}
